pkg/net/cidrs/triemap.go

```diff
@@ -197,10 +213,12 @@ func (t *trieMap) getIPv4(addr netip.Addr) (int, bool) {
 	}
 	// walk IP bits high to low, checking if current node matches
 	ip := addr.As4()
+	// first cast to uint32 for fast bit access
+	// NOTE: IP addresses are big endian, so the low bits are in the last byte
 	ipInt := uint32(ip[3]) | uint32(ip[2])<<8 | uint32(ip[1])<<16 | uint32(ip[0])<<24
 	for i := 31; i >= 0; i-- {
 		// walk based on current address bit
-		if (ipInt>>i)&1 == 1 {
+		if (ipInt & (uint32(1) << i)) != 0 {
```

Comment:
(comparing to zero is a fast built-in operation on all? CPU, the compiler can probably? optimize these predictable bitshifts, and we have to do one mask either way; benchmarks seem to agree)